build.rs

```diff
@@ -1,13 +1,79 @@
 use std::env;
 
 fn find_library_paths() -> Vec<String> {
-    match env::var("CUDA_LIBRARY_PATH") {
-        Ok(path) => {
-            let split_char = if cfg!(target_os = "windows") { ";" } else { ":" };
+    if let Ok(path) = env::var("CUDA_LIBRARY_PATH") {
+        // The location of the libcuda, libcudart, and libcublas can be hardcoded with the
+        // CUDA_LIBRARY_PATH environment variable.
+        let split_char = if cfg!(target_os = "windows") {
+            ";"
+        } else {
+            ":"
+        };
 
-            path.split(split_char).map(|s| s.to_owned()).collect::<Vec<_>>()
+        path.split(split_char)
+            .map(|s| s.to_owned())
+            .collect::<Vec<_>>()
+    } else if let Ok(path) = env::var("CUDA_PATH") {
+        // If CUDA_LIBRARY_PATH is not found, then CUDA_PATH will be used when building for Windows
+        // to locate the Cuda installation. Cuda installs the full Cuda SDK for 64-bit, but only
+        // a limited set of libraries for 32-bit. Namely, it does not include cublas in 32-bit,
+        // which cuda-sys requires.
+
+        // 'path' points to the base of the CUDA Installation. The lib directory is a sub-directory.
+        let path = std::path::Path::new(&path);
+
+        // To do this the right way, we check to see which target we're building for.
+        let target = env::var("TARGET")
+            .expect("cargo did not set the TARGET environment variable as required.");
```

Comment:
A blunt `cfg!(target_os = "windows")` for the whole CUDA_PATH branch is probably fine, but we need to read `TARGET` to support cross-compiling for 32-bit Windows (and WoW) from the 64-bit Windows since build scripts are always compiled for the native architecture, even when cross-compiling.

Reply:
But cross-compiling is otherwise currently blocked by #9 (Nvidia does not ship 32-bit libraries for cublas)

Reply:
I added a check for `cfg!(target_os = "windows")` before checking for `CUDA_PATH`.